### PR TITLE
Update of handsontable

### LINKED ISSUE #84: Latest version (18-08-2015) has unwanted side effect on notifications
Updating the edittable plugin to the latest version (18-08-2015) in my dokuwiki (both Hrun as Detritus) has the unwanted side effect that notifications become completely filled with icons, like shown below.

schermafbeelding 2015-08-25 om 16 30 43

becomes

schermafbeelding 2015-08-25 om 16 30 00

I've verified that switching off the edittable plugin removes this side effect.

### LINKED ISSUE #89: Table in editmode does not fit on page
If I open a table in edit mode, and there are some columns with more text into it, the table does not fit into the page width. Instead of this, the right columns disappear. 
If I drag the columns smaller, than word-wrap is applied to the columns with more text and it is possible to adjust the table to fit into page width.

I would prefer, that the table opens with word wrap and tries to auto-adjust the columns size to fit into the page. I think, this was also the normal behavior before the last update.

screenshot

### LINKED ISSUE #102: insert rows from excel
when inserting cells from excel and the there are not enough cells to fit, the following error is thrown

```
TypeError: meta[row] is undefined
return meta[row][col]; 
```

in handsonetable the table rows/cols will be adjusted to fit the copied cells

### LINKED ISSUE #105: move to table after save/cancel
after saving page in text editor you get to the right section.
is this possible for edittable too?

### LINKED ISSUE #109: action "remove row" on thead row
the action "remove row" on the thead row (A,B,C,D ...) is possible, but makes no sense and results in a Js error.

### PR DESCRIPTION
**Pull request by micgro42**
Update the underlying handsontable from 0.16.0 to the current version of 0.34.0

#### Primary goal of this pull request
- [x] copying from/to spreadsheets works
  - [x] copying from/to spreadsheets works if the copied data is larger than the table (fixes #102)
#### secondary bugfixes
- [x] Ensure that we scroll to the correct position in the page after finishing editing the table (fixes #105) _(works for me, I consider it fixed)_
- [x] check performance of large tables (fixes #89) _(cannot reproduce. I consider it fixed)_
- [x] check issue with disappeaing rows (#94) _(cannot reproduce. I consider it fixed)_
- [x] scrolling with mouse-wheel works with acceptable speed
- [x] delete row/col must not be possible on 0th col/row (fixes #109 )

##### core functionality
- [x] Extend table downwards when doing autofill
- [x] Extend table to the right when doing autofill
- [x] Rows/columns cannot be moved within joined rows/columns
- [x] adding rows/columns works as expected in tables with joined rows/columns
- [x] joining cells
- [x] splitting joined cells
- [x] Linebreaks are converted to DokuWiki-linebreak-syntax (`\\`) _works, but #134 is still unresolved_
##### dokuwiki peripherical
- [x] Ensure linkwizard is usable, c.f. 685500035501a540905e60b4573421b714a54bb4
- [x] Ensure the other toolbar pickers work as well
##### styling
- [x] context menu is correctly styled
- [x] there is no stray `@charset` in the styles, c.f. 918e16209e28050b3c25b3ae3d812d039d45871a  (was already fixed, fixes #84 )
- ...

~~### Dependencies~~
~~**The plugin now depends also on the [jspolyfill-plugin](https://dokuwiki.org/plugin:jspolyfill) for IE11 support**~~

### CODE REVIEW

File: script/editor.js [hunk 1]
```diff
@@ -2,56 +2,37 @@ var edittable = edittable || {};
...
- * @param start int
- * @param end int optional, only for moves
- */
-edittable.updateMergeInfo = function (direction, type, start, end) {
-    var mergesNeedUpdate = false;
-    if (type === 'create' || type === 'remove') {
-        end = Infinity;
-    }
+    'use strict';
+
```
Comment (splitbrain) [inline script/editor.js]: Could we have doc blocks for these functions?

File: script/editor.js [hunk 2]
```diff
@@ -2,56 +2,37 @@ var edittable = edittable || {};
...
-    }
-};
+
+    edittable.moveCol = function moveCol(movingColIndexes, target, dmarray) {
+        return dmarray.map(function (row) {
+            return edittable.moveRow(movingColIndexes, target, row);
+        });
+    };
 
 edittable.getMerges = function (meta) {
```
Comment (splitbrain) [inline script/editor.js]: this one seems to be indented wrong now.

**Comment by splitbrain** [comment]
I'm not sure I like the dependency on the jspolyfill plugin. As far as I can see jspolyfill is only used by this plugin currently and it only polfills `array.fill` -- that one polyfill could easily be included into this plugin (it wouldn't matter much sizewise). I guess we need to discuss the future use of the polyfill plugin.

**Comment by splitbrain** [comment]
- [ ] update deleted.files
- [ ] Adjust README.handsontable

**Comment by splitbrain** [comment]
`less/handsontable.less` is basically the original `handsontable.css` file, thus I would find it logical to move it to the `lib` directory. However it seems there are some adjustments made by us:

https://github.com/cosmocode/edittable/blob/e07451101e6df50c6264b09320a7ccd281b318ed/less/handsontable.less#L29-L34

I would split those out into their own file (kept in the `less/` directory).

**Comment by splitbrain** [comment]
I added `eslint` in another branch. I guess it would replace `jshint`. Not sure about the rules. I like that it checks the docblocks...

**Comment by micgro42** [comment]
Moving the `less/handsontable.less` to the `lib` directory is a good idea.
However the adjustments are not ours, but from upstream:
https://github.com/handsontable/handsontable/blob/b6a9952246b059218214ad4fad4d4de6e0f5a03c/dist/handsontable.full.css#L29-L34

**Pull request merged**